Studio-Lovelies/GG-JointJustice-Unity
# Automated bug tracking via Sentry

## Summary
Implements what I've ~~threatened to do~~ suggested in https://github.com/Studio-Lovelies/GG-JointJustice-Unity/discussions/199.

Due to #320 still waiting on team approval, I have created a shared GGJJ account in the meantime.  
For info on how to access it, simply ping me on Discord.

Also fixes a few issues inside the docs_generator.

## Before/after screenshots and/or animated gif
### During development...
...nothing will change. Sentry is disabled in the Editor by default and the credentials aren't even pushed to this repository.
If you want to try it out, sign up at [sentry.io](https://sentry.io/), create a new project, and Sentry will generate a `DSN` for you.  
Enter it via the Unity Editor and enable capturing in the editor.
![image](https://user-images.githubusercontent.com/1689033/215293647-19460cf9-47da-42c9-bd7a-d0fa2fd051ee.png)
Encounter an error (or force one by throwing an exception somewhere in the code) and visit [sentry.io](https://sentry.io/) again. You will notice info like a stack trace, host OS, and versions of assemblies about the error immediately after seeing it pop up in the Unity Editor log window.

### During regular gameplay...
...nothing visible will change either.  
The game automatically traces and sends info about runtime errors to the shared GGJJ Sentry account in the background.

## Testing instructions
1. Be aware of any way to force a call to `Debug.LogError` or exception in `develop` without modifying code (I haven't found any yet!)
2. Download a build of this branch
3. Run into that error
4. Log into [sentry.io](https://sentry.io/) with the shared GGJJ account
5. See information about the error

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  - implements #199
